### PR TITLE
various fixes

### PR DESCRIPTION
**Pull request by anoopsharma00**
-- Implicit xns for ddl stmts are now started in master exe if
   autocommit is OFF
-- default values for varchar columns are now handled correctly
-- DATEFORMAT function on timestamp values are now handled correctly
-- CREATE/DROP PROCEDURE stmt are now handled correctly with DDL xns.
-- alter primary key stmt now returns an error if table already has a
   user defined primary/clustering key
-- regress/core/TEST116, seabase/TEST030,031 have been updated with new tests

### CODE REVIEW

**Comment by anoopsharma00** [comment]
jenkins retest

**Comment by svarnau** [comment]
Anoop, your commits almost always come from centos user.  That stays in the code history forever. It would be helpful that you could set your git user info when you set up new VM, before making commits. Thanks.

**Comment by selvaganesang** [comment]
Can you please link [TRAFODION-1864] LONGWVARCHAR column support have problems with storage of the column in hbase and uncomment the tests in charsets/TEST012

**Comment by anoopsharma00** [comment]
With the code fix, this charsets/TEST012 now passes without any known differences.
Is there something else that need to be uncommented in TEST012? Or did you mean
to remove the known diff file?

**Comment by anoopsharma00** [comment]
varchar default values pbm is being tracked by JIRA [TRAFODION-1864]

File: core/sql/generator/GenExpGenerator.cpp [hunk 1]
```diff
@@ -753,12 +752,42 @@ void ExpGenerator::copyDefaultValues(
...
-	  str_cpy_all(tgtDefVal, 
-		      srcAttr->getDefaultValue(), 
-		      srcAttr->getDefaultValueStorageLength());
+          Lng32 tgtDefLen = tgtAttr->getDefaultValueStorageLength();
+          Lng32 srcDefLen = srcAttr->getDefaultValueStorageLength();
+          char* srcDefVal = srcAttr->getDefaultValue();
+	  char * tgtDefVal = new(generator->getSpace()) char[tgtDefLen];
+          // if source and target def storage lengths dont match, then
+          // need to move each part (null, vclen, data) separately.
+          if (tgtDefLen != srcDefLen)
```
Comment (DaveBirdsall) [inline core/sql/generator/GenExpGenerator.cpp]: Might we be fooled into thinking the tgtDefLen is equal to the srcDefLen, but src is nullable varchar(n-2) while tgt is non-nullable varchar(n)?

**Comment by selvaganesang** [comment]
Thanks Anoop. Looks like I created the known diff file. If charsets/TEST012 passes without the diff file, then the issue is fixed

**Comment by DaveBirdsall** [comment]
With one possible exception (see inline comment), the changes look good to me.

**Comment by anoopsharma00** [comment]
jenkins retest

**Pull request merged**